src/upscale.rs

```diff
@@ -102,21 +102,36 @@ impl TestnetDeployer {
             ];
             self.plan(
                 Some(vars),
-                options.current_inventory.environment_type.clone(),
+                options
+                    .current_inventory
+                    .environment_details
+                    .environment_type
+                    .clone(),
             )
             .await?;
             return Ok(());
         }
 
         self.create_or_update_infra(
             &options.current_inventory.name,
+            Some(
+                match options
+                    .current_inventory
+                    .environment_details
+                    .deployment_type
+                {
+                    DeploymentType::New => 1,
+                    DeploymentType::Bootstrap => 0,
+                },
+            ),
             Some(desired_auditor_vm_count),
             Some(desired_bootstrap_node_vm_count),
             Some(desired_node_vm_count),
             Some(desired_uploader_vm_count),
```

Comment:
Uploaders cannot be upscaled for a `DeploymentType::Bootstrap` environment as we will not have a facuet to get tokens from. Maybe throw an error if count > 0 & type = bootstrap?

Reply:
Also, I can see that `bootstrap.rs` sets `auditor vm count = 0` & `bootstrap node vm count =0` but the upscale command can potentially make them > 0.

So some assumptions like the below can be invalidated.
```
        // If there are no bootstrap nodes, it's a bootstrap deploy, and in that case, we're not
        // really interested in available peers.
```

Reply:
I think the changes I've pushed should address the concerns here. Thanks.